plugins/org.python.pydev/src/org/python/pydev/builder/flake8/Flake8PrefPage.java

```diff
@@ -27,19 +33,30 @@
     private RadioGroupFieldEditor searchFlake8Location;
     private FileFieldEditor fileField;
     private List<FieldEditor> fields = new ArrayList<FieldEditor>(5);
+    private JsonFieldEditor jsonFieldEditor;
 
     public static final int COLS = 4;
 
-    public static final String[][] LABEL_AND_VALUE = new String[][] {
-            { "Error", String.valueOf(IMarker.SEVERITY_ERROR) },
-            { "Warning", String.valueOf(IMarker.SEVERITY_WARNING) },
-            { "Info", String.valueOf(IMarker.SEVERITY_INFO) },
-            { "Ignore", String.valueOf(Flake8Preferences.SEVERITY_IGNORE) }, };
-
     public static final String[][] SEARCH_FLAKE8_LOCATION_OPTIONS = new String[][] {
             { "Search in interpreter", Flake8Preferences.LOCATION_SEARCH },
             { "Specify Location", Flake8Preferences.LOCATION_SPECIFY },
     };
+    private static final String CODES_CONFIG_TEMPLATE = ""
+            + "{\n"
+            + "    \"C\": \"error\",\n"
+            + "    \"E\": \"warning\",\n"
+            + "    \"F\": \"info\",\n"
+            + "    \"W\": \"ignore\",\n"
+            + "    \"C200\": 2,\n"
```

Comment:
I think we should just show the variants of `"error", "warning", "info" and "ignore" (the variants related to the numbers don't need to be shown)... we can still support it, just don't show in the template.

Also, I think that we don't need the button to add this... just make this the default (in `DEFAULT_FLAKE8_CODES_CONFIG`) -- so, the button can be removed.

I think that we should use:

```
""
            + "{\n"
            + "    \"E\": \"error\",\n"
            + "    \"W\": \"warning\",\n"
            + "    \"C9\": \"info\",\n"
            + "    \"N8\": \"warning\",\n"
            + "    \"F\": \"warning\",\n"
            + "    \"E[400,500]\": \"error\",\n"
            + "    \"E600\": \"error\",\n"
            + "}"
            + "";
```

as the default there (it has multiple duplicates to the `E` to show error, but that should be ok (those would mostly be examples of how to configure and it shouldn't make much difference in practice).